libs/qCC_io/LASFilter.cpp

```diff
@@ -32,1408 +32,1167 @@
+				pDlg->setValue(pDlg->value() + 1);
 			}
+			QApplication::processEvents();
 		}
 
-		if (parameters.alwaysDisplayLoadDialog && !s_lasOpenDlg->autoSkipMode() && !s_lasOpenDlg->exec())
+		viewSet = reader.result();
+		pointView = *viewSet.begin();
+
+		if (parameters.parentWidget)
 		{
-			ifs.close();
-			return CC_FERR_CANCELED_BY_USER;
+			pDlg.reset(new ccProgressDialog(true, parameters.parentWidget)); //cancel available
+			pDlg->setMethodTitle(QObject::tr("Tiling points"));
+			pDlg->setInfo(QObject::tr("Points: %L1").arg( nbOfPoints ));
+			pDlg->start();
 		}
-		bool ignoreDefaultFields = s_lasOpenDlg->ignoreDefaultFieldsCheckBox->isChecked();
+		CCLib::NormalizedProgress nProgress(pDlg.data(), nbOfPoints);
 
-		TilingStruct tiler;
-		bool tiling = s_lasOpenDlg->tileGroupBox->isChecked();
-		if (tiling)
+		for (PointId idx = 0; idx < pointView->size(); ++idx)
 		{
-			//tiling (vertilca) dimension
-			unsigned vertDim = 2;
-			switch (s_lasOpenDlg->tileDimComboBox->currentIndex())
-			{
-			case 0: //XY
-				vertDim = 2;
-				break;
-			case 1: //XZ
-				vertDim = 1;
-				break;
-			case 2: //YZ
-				vertDim = 0;
-				break;
-			default:
-				assert(false);
-				break;
-			}
-
-			unsigned w = static_cast<unsigned>(s_lasOpenDlg->wTileSpinBox->value());
-			unsigned h = static_cast<unsigned>(s_lasOpenDlg->hTileSpinBox->value());
-
-			QString outputBaseName = s_lasOpenDlg->outputPathLineEdit->text() + "/" + QFileInfo(filename).baseName();
-
-			if (!tiler.init(w, h, vertDim, outputBaseName, bbMin, bbMax, header))
-			{
-				//failed to open at least one file!
-				return CC_FERR_WRITING;
-			}
+			if (pDlg->isCancelRequested())
+				return CC_FERR_CANCELED_BY_USER;
+			tiler.addPoint(pointView, idx);
+			nProgress.oneStep();
 		}
 
-		//RGB color
-		liblas::Color rgbColorMask; //(0,0,0) on construction
-		if (s_lasOpenDlg->doLoad(LAS_RED))
-			rgbColorMask.SetRed(~0);
-		if (s_lasOpenDlg->doLoad(LAS_GREEN))
-			rgbColorMask.SetGreen(~0);
-		if (s_lasOpenDlg->doLoad(LAS_BLUE))
-			rgbColorMask.SetBlue(~0);
-		bool loadColor = (rgbColorMask[0] || rgbColorMask[1] || rgbColorMask[2]);
-
-		//progress dialog
-		QScopedPointer<ccProgressDialog> pDlg(0);
+		// Now the tiler will actually write the points
+		QFuture<void> writer = QtConcurrent::run([&tiler]() {tiler.writeAll(); });
 		if (parameters.parentWidget)
 		{
-			pDlg.reset(new ccProgressDialog(true, parameters.parentWidget)); //cancel available
-			pDlg->setMethodTitle(QObject::tr("Open LAS file"));
-			pDlg->setInfo(QObject::tr("Points: %L1").arg( nbOfPoints ));
+			pDlg.reset(new ccProgressDialog(false, parameters.parentWidget));
+			pDlg->setMethodTitle(QObject::tr("LAS file"));
+			pDlg->setInfo(QObject::tr("Please wait... writing in progress"));
+			pDlg->setRange(0, 0);
+			pDlg->setModal(true);
 			pDlg->start();
 		}
-		CCLib::NormalizedProgress nProgress(pDlg.data(), nbOfPoints);
+		while (!writer.isFinished())
+		{
+#if defined(CC_WINDOWS)
```

Comment:
Same comment as above - Sleep seems strange. Can it be avoided in favour of `writer.waitForFinished()` somehow?